# Amazon Report

In [4]:
import pandas as pd
import numpy as np

In [5]:
customers = pd.read_csv("amazon_customers_dirty.csv")
deliveries = pd.read_csv("amazon_deliveries_dirty.csv")
order_items = pd.read_csv("amazon_order_items_dirty.csv")
orders = pd.read_csv("amazon_orders_dirty.csv")
products= pd.read_csv("amazon_products_dirty.csv")
stocks= pd.read_csv("amazon_stock_dirty.csv")

# Clean the customers data

In [6]:
customers.head(10)

customer_id      customer_name                        email      country  \
0       C1000  Alexandra Carlson  alexandra.carlson@gmail.com         Chad   
1       C1001        Donna Evans        donna.evans@gmail.com  Afghanistan   
2       C1002      Kimberly Levy    kimberly.levy@outlook.com       monaco   
3       C1003     Jennifer Wells     jennifer.wells@yahoo.com     PARAGUAY   
4       C1004        Erin Garcia        erin.garcia@yahoo.com    Nicaragua   
5       C1005      Jonathon Hood      jonathon.hood@yahoo.com     Cambodia   
6       C1006        Jeffery Lam                          NaN         Togo   
7       C1007    Roberta Gardner   roberta.gardner@icloud.com     Colombia   
8       C1008       Ralph Dillon       ralph.dillon@gmail.com        india   
9       C1009       John Acevedo      john.acevedo@icloud.com        Benin   

  registration_date  
0         11/3/2021  
1         8/26/2020  
2         6/27/2014  
3        11/11/2024  
4          2/2/2024  
5         2/27/2019  
6        11/15/2016  
7         4/14/2022  
8         10/4/2017  
9         4/16/2024

In [8]:
customers.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   customer_id        1000 non-null   object
 1   customer_name      1000 non-null   object
 2   email              950 non-null    object
 3   country            970 non-null    object
 4   registration_date  1000 non-null   object
dtypes: object(5)
memory usage: 39.2+ KB


In [10]:
customers.isnull().sum()

customer_id           0
customer_name         0
email                50
country              30
registration_date     0
dtype: int64

In [14]:
customers_clean = customers.copy()

In [16]:
customers_clean["country"] = customers_clean["country"].fillna("Unknow")

In [17]:
customers_clean["has_email"] = customers_clean["email"].notnull().astype(int)

In [18]:
customers_clean.isnull().sum()

customer_id           0
customer_name         0
email                50
country               0
registration_date     0
has_email             0
dtype: int64

In [19]:
customers_clean.head()

customer_id      customer_name                        email      country  \
0       C1000  Alexandra Carlson  alexandra.carlson@gmail.com         Chad   
1       C1001        Donna Evans        donna.evans@gmail.com  Afghanistan   
2       C1002      Kimberly Levy    kimberly.levy@outlook.com       monaco   
3       C1003     Jennifer Wells     jennifer.wells@yahoo.com     PARAGUAY   
4       C1004        Erin Garcia        erin.garcia@yahoo.com    Nicaragua   

  registration_date  has_email  
0         11/3/2021          1  
1         8/26/2020          1  
2         6/27/2014          1  
3        11/11/2024          1  
4          2/2/2024          1

In [21]:
customers_clean.sample(10)

customer_id    customer_name                        email  \
536       C1536     Denise Brown       denise.brown@yahoo.com   
989       C1989     Sandra Davis       sandra.davis@yahoo.com   
632       C1632     Mary Russell     mary.russell@outlook.com   
173       C1173      Julian Rose       julian.rose@icloud.com   
934       C1934  Douglas Holland  douglas.holland@hotmail.com   
613       C1613    Debra Brady         debra.brady@icloud.com   
769       C1769        John Wood          john.wood@yahoo.com   
875       C1875    Natalie Davis    natalie.davis@outlook.com   
407       C1407       Todd Ortiz                          NaN   
917       C1917   David Anderson   david.anderson@hotmail.com   

                         country registration_date  has_email  
536                       Kuwait         1/23/2020          1  
989                    Hong Kong          6/4/2020          1  
632                   Montserrat          1/9/2019          1  
173                       Turkey         10/5/2024          1  
934  Falkland Islands (Malvinas)         7/19/2017          1  
613                       Guyana          8/8/2020          1  
769                      Bahamas         4/14/2019          1  
875                      Armenia        11/21/2017          1  
407                       Jordan          7/8/2021          0  
917                        japan          9/8/2023          1

# Clean the Deliveries Data

In [11]:
deliveries.head(10)

order_id   ship_date delivery_date delivery_status           courier
0   O10000  2025-01-04    2025-01-07         On Time  Amazon Logistics
1   O10001  2025-01-03    2025-01-05            Late  Amazon Logistics
2   O10003  2025-01-02    2025-01-04            Late             FedEx
3   O10004  2025-01-02    2025-01-06         On Time               UPS
4   O10005  2025-01-04           NaN            Late             FedEx
5   O10009  2025-01-03    2025-01-05         On Time               DHL
6   O10010  2025-01-04    2025-01-09         On Time               UPS
7   O10013  2025-01-02    2025-01-04            Late               DHL
8   O10014  2025-01-02    2025-01-05         On Time               DHL
9   O10019  2025-01-02    2025-01-04         On Time               DHL

In [12]:
deliveries.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7254 entries, 0 to 7253
Data columns (total 5 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   order_id         7254 non-null   object
 1   ship_date        7254 non-null   object
 2   delivery_date    6529 non-null   object
 3   delivery_status  7254 non-null   object
 4   courier          7254 non-null   object
dtypes: object(5)
memory usage: 283.5+ KB


In [13]:
deliveries.isnull().sum()

order_id             0
ship_date            0
delivery_date      725
delivery_status      0
courier              0
dtype: int64

###### Found null value in delivery_date

In [26]:
deliveries_clean = deliveries.copy() # make a copy of the orginal dirty data

deliveries_clean["delivered"] = deliveries_clean["delivery_date"].notnull().astype(int) #add a delivered flag

In [27]:
deliveries_clean.isnull().sum()

order_id             0
ship_date            0
delivery_date      725
delivery_status      0
courier              0
delivered            0
dtype: int64

In [30]:
deliveries_clean.sample(10)

order_id   ship_date delivery_date delivery_status           courier  \
1758   O13539  2025-02-14    2025-02-16         On Time  Amazon Logistics   
2936   O15981  2025-03-15    2025-03-19            Late             FedEx   
3804   O17674  2025-04-06    2025-04-10            Late               UPS   
4117   O18312  2025-04-15    2025-04-19            Late               DHL   
6609   O23359  2025-06-16           NaN            Late  Amazon Logistics   
173    O10352  2025-01-05    2025-01-07         On Time               DHL   
4704   O19468  2025-04-29    2025-05-01            Late             FedEx   
667    O11346  2025-01-19    2025-01-21            Late               DHL   
6353   O22834  2025-06-10    2025-06-15            Late               UPS   
5935   O21969  2025-05-30    2025-06-02            Late               DHL   

      delivered  
1758          1  
2936          1  
3804          1  
4117          1  
6609          0  
173           1  
4704          1  
667           1  
6353          1  
5935          1

# Clean Order_items Data

In [40]:
order_items.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 44004 entries, 0 to 44003
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   order_id     44004 non-null  object 
 1   product_id   44004 non-null  object 
 2   quantity     44004 non-null  int64  
 3   unit_price   44004 non-null  float64
 4   total_price  44004 non-null  float64
 5   discount     42244 non-null  float64
dtypes: float64(3), int64(1), object(2)
memory usage: 2.0+ MB


In [31]:
order_items.isnull().sum()

order_id          0
product_id        0
quantity          0
unit_price        0
total_price       0
discount       1760
dtype: int64

##### found null value at discount

In [32]:
order_items_clean = order_items.copy()

In [33]:
order_items_clean["discount"] = order_items_clean["discount"].fillna(0.0)

In [34]:
order_items_clean.isnull().sum()

order_id       0
product_id     0
quantity       0
unit_price     0
total_price    0
discount       0
dtype: int64

In [35]:
order_items_clean.sample(10)

order_id product_id  quantity  unit_price  total_price  discount
40332   O14159      P1031         5      113.46       425.48      0.25
40854   O16698      P1262         5      145.92       642.05      0.12
4003    O12673      P1201         2      249.67       389.49      0.22
27055   O17487      P1014         1       23.47        20.18      0.14
769     O12672      P1150         3      164.09       403.66      0.18
19188   O20927      P1191         5      264.47       938.87      0.29
11414   O13161      P1055         5       56.75       258.21      0.09
3297    O14921      P1246         3       64.56       162.69      0.16
42358   O13676      P1232         3      118.71       349.01      0.02
17315   O14672      P1084         1      181.36       157.78      0.13

# Clean Orders Data

In [39]:
orders.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14668 entries, 0 to 14667
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   order_id      14668 non-null  object
 1   customer_id   14668 non-null  object
 2   order_date    14668 non-null  object
 3   order_status  14228 non-null  object
dtypes: object(4)
memory usage: 458.5+ KB


In [36]:
orders.isnull().sum()

order_id          0
customer_id       0
order_date        0
order_status    440
dtype: int64

##### Found null value in order_status

In [60]:
orders_clean = orders.copy()

In [61]:
orders_clean["order_status"] = orders_clean["order_status"].fillna("Unknow")

In [62]:
orders_clean.isnull().sum()

order_id        0
customer_id     0
order_date      0
order_status    0
dtype: int64

In [69]:
orders_clean.sample(10)

order_id customer_id  order_date order_status
8933    O18933       C1133  20-04-2025    Cancelled
7947    O17947       C1596  07-04-2025    Cancelled
6122    O16122       C1078  15-03-2025       Placed
4073    O14073       C1290  19-02-2025      Shipped
965     O10965       C1508  12-01-2025       Unknow
4020    O14020       C1347  18-02-2025       Placed
299     O10299       C1441  03-01-2025      Shipped
1458    O11458       C1915  18-01-2025    Cancelled
7359    O17359       C1877  30-03-2025      Shipped
10849   O20849       C1436  15-05-2025       Placed

# Clean the Products Data

In [70]:
products.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 300 entries, 0 to 299
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   product_id    300 non-null    object 
 1   product_name  300 non-null    object 
 2   category      300 non-null    object 
 3   unit_cost     295 non-null    float64
 4   unit_price    300 non-null    object 
dtypes: float64(1), object(4)
memory usage: 11.8+ KB


In [71]:
products.isnull().sum()

product_id      0
product_name    0
category        0
unit_cost       5
unit_price      0
dtype: int64

In [72]:
products.head()

product_id       product_name         category  unit_cost unit_price
0      P1000       Trouble Lite           Health      40.58     285.71
1      P1001         Rich Lite   Office Supplies      74.54     183.61
2      P1002        Without Kit           Sports      19.82      55.24
3      P1003  International Max  Office Supplies      10.52     261.19
4      P1004           Race Pro          Fashion        NaN     215.34

In [73]:
products_clean = products.copy()

In [75]:
products_clean["unit_cost"] = products_clean["unit_cost"].fillna(0.0)

In [76]:
products_clean.isnull().sum()

product_id      0
product_name    0
category        0
unit_cost       0
unit_price      0
dtype: int64

# Clean the Stock Data

In [84]:
stocks.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 4 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   product_id          1000 non-null   object 
 1   warehouse_location  1000 non-null   object 
 2   stock_level         1000 non-null   float64
 3   last_updated        1000 non-null   object 
dtypes: float64(1), object(3)
memory usage: 31.4+ KB


In [85]:
stocks.isnull().sum()

product_id            0
warehouse_location    0
stock_level           0
last_updated          0
dtype: int64

In [93]:
stocks.head()

product_id warehouse_location  stock_level last_updated
0      P1076              Tokyo        284.0   2025-03-03
1      P1062             Berlin        329.0   2025-01-17
2      P1289              Tokyo          0.0   2025-03-04
3      P1010                NYC         26.0   2025-02-16
4      P1227                NYC        136.0   2025-05-26

In [94]:
stocks_clean = stocks.copy()

In [95]:
stocks_clean["stock_level"] = stocks_clean["stock_level"].fillna(0.0)

In [96]:
stocks_clean.isnull().sum()

product_id            0
warehouse_location    0
stock_level           0
last_updated          0
dtype: int64

In [97]:
stocks_clean.sample(5)

product_id warehouse_location  stock_level last_updated
960      P1158             Berlin        165.0   2025-06-08
65       P1144              Delhi        438.0   2025-06-05
69       P1186             Berlin        148.0   2025-04-24
798      P1034              Tokyo        114.0   2025-05-26
868      P1018             Berlin        276.0   2025-03-27

# Identify Customer Segments to Target

### Which customer segments should we focus on for this campaign?

##### Join the Data Set  to get Full Customer Behavior 

In [98]:
orders_full = pd.merge(orders_clean, customers_clean, on="customer_id", how="inner")

In [99]:
orders_full = pd.merge(orders_full, order_items, on="order_id", how= "inner")

In [100]:
orders_full.head()

order_id customer_id  order_date order_status  customer_name  \
0   O10000       C1198  01-01-2025    Delivered  Robert Arnold   
1   O10000       C1198  01-01-2025    Delivered  Robert Arnold   
2   O10000       C1198  01-01-2025    Delivered  Robert Arnold   
3   O10001       C1359  01-01-2025      Shipped  Timothy Davis   
4   O10001       C1359  01-01-2025      Shipped  Timothy Davis   

                      email  country registration_date  has_email product_id  \
0   robert.arnold@yahoo.com  Lesotho         10/6/2020          1      P1204   
1   robert.arnold@yahoo.com  Lesotho         10/6/2020          1      P1248   
2   robert.arnold@yahoo.com  Lesotho         10/6/2020          1      P1015   
3  timothy.davis@icloud.com  Romania         8/13/2013          1      P1125   
4  timothy.davis@icloud.com  Romania         8/13/2013          1      P1030   

   quantity  unit_price  total_price  discount  
0         3       36.47       102.85      0.06  
1         1       32.54        24.08      0.26  
2         5       59.45       297.25      0.00  
3         5      121.68       504.97      0.17  
4         1       88.69        80.71      0.09

##### Calulate the RFM Metrics for Each Customer
###### RFM = Recency, Frequency, Monetary

In [102]:
from datetime import datetime

#Convert order_date to datetime
orders_full["order_date"] = pd.to_datetime(orders_full["order_date"], dayfirst=True)

#Define the fixed reference date
ref_date = datetime(2025, 7, 31)

rfm = orders_full.groupby("customer_id").agg({
                          "order_date" : lambda x:(ref_date - x.max()).days,
                          "order_id" : "count",
                          "total_price": "sum"
}).reset_index()

rfm.columns= ["customer_id", "recency", "frequency", "monetary"]

##### Create RFM Segments

In [103]:
rfm["r_score"] = pd.qcut(rfm["recency"], 4, labels=[4, 3, 2, 1])
rfm["f_score"] = pd.qcut(rfm["frequency"].rank(method="first"), 4, labels=[1, 2, 3, 4])
rfm["m_score"] = pd.qcut(rfm["monetary"], 4, labels=[1, 2, 3, 4])

rfm["rfm_score"] = rfm["r_score"].astype(str) + rfm["f_score"].astype(str) + rfm["m_score"].astype(str)

In [104]:
rfm[['customer_id', 'recency', 'frequency', 'monetary', 'r_score', 'f_score', 'm_score', 'rfm_score']]


customer_id  recency  frequency  monetary r_score f_score m_score  \
0         C1000       52         52  19370.18       1       3       3   
1         C1001       42         45  15973.26       2       3       3   
2         C1002       35         70  28596.55       4       4       4   
3         C1003       38         69  26206.11       3       4       4   
4         C1004       32         37  14422.93       4       2       2   
..          ...      ...        ...       ...     ...     ...     ...   
995       C1995       33         43  22114.24       4       2       4   
996       C1996       56         16   4379.61       1       1       1   
997       C1997       43         54  21440.96       2       4       4   
998       C1998       47         43  14896.24       2       3       2   
999       C1999       43         49  19125.50       2       3       3   

    rfm_score  
0         133  
1         233  
2         444  
3         344  
4         422  
..        ...  
995       424  
996       111  
997       244  
998       232  
999       233  

[1000 rows x 8 columns]

In [106]:
# Top customers: RFM score = 444
rfm[rfm['rfm_score'] == '444']


customer_id  recency  frequency  monetary r_score f_score m_score  \
2         C1002       35         70  28596.55       4       4       4   
12        C1012       33         55  22461.79       4       4       4   
23        C1023       33         56  22710.01       4       4       4   
24        C1024       33         65  19676.96       4       4       4   
41        C1041       31         55  20517.89       4       4       4   
..          ...      ...        ...       ...     ...     ...     ...   
939       C1939       33         56  23214.86       4       4       4   
957       C1957       31         57  22343.11       4       4       4   
974       C1974       32         63  22784.50       4       4       4   
980       C1980       31         70  23097.37       4       4       4   
981       C1981       33         77  27065.98       4       4       4   

    rfm_score  
2         444  
12        444  
23        444  
24        444  
41        444  
..        ...  
939       444  
957       444  
974       444  
980       444  
981       444  

[81 rows x 8 columns]

In [107]:
def assign_segment(rfm_row):
    if rfm_row['rfm_score'] == '444':
        return 'Champions'
    elif rfm_row['r_score'] == '4':
        return 'Recent Customers'
    elif rfm_row['f_score'] == '4':
        return 'Frequent Buyers'
    elif rfm_row['m_score'] == '4':
        return 'Big Spenders'
    elif rfm_row['r_score'] == '1':
        return 'Lost Customers'
    else:
        return 'Others'

# Apply segment labeling
rfm['segment'] = rfm.apply(assign_segment, axis=1)

In [108]:
rfm['segment'].value_counts()


segment
Others       919
Champions     81
Name: count, dtype: int64

In [109]:
target_customers = rfm[rfm['segment'] == 'Champions']


In [110]:
print(target_customers)

    customer_id  recency  frequency  monetary r_score f_score m_score  \
2         C1002       35         70  28596.55       4       4       4   
12        C1012       33         55  22461.79       4       4       4   
23        C1023       33         56  22710.01       4       4       4   
24        C1024       33         65  19676.96       4       4       4   
41        C1041       31         55  20517.89       4       4       4   
..          ...      ...        ...       ...     ...     ...     ...   
939       C1939       33         56  23214.86       4       4       4   
957       C1957       31         57  22343.11       4       4       4   
974       C1974       32         63  22784.50       4       4       4   
980       C1980       31         70  23097.37       4       4       4   
981       C1981       33         77  27065.98       4       4       4   

    rfm_score    segment  
2         444  Champions  
12        444  Champions  
23        444  Champions  
24        444  

### Save the result

In [111]:
# Save to CSV
target_customers.to_csv('champion_customers_report.csv', index=False)


Based on our analysis of Amazon customer purchase behavior and segmentation, we have identified key customer segments and tailored product offer strategies designed to maximize engagement and campaign effectiveness.

Champions are our most valuable customers—they purchase frequently, spend the most, and have recently interacted with our platform. Their top product interests include Electronics, Health products, and Office Supplies. For this segment, we recommend offering exclusive early access to premium or new product launches, alongside loyalty rewards programs to deepen their engagement and encourage continued high-value purchases.

Frequent Buyers regularly purchase across various categories but typically at moderate spending levels. Their preferred categories include Office Supplies and Household Essentials. Campaigns targeting this group should focus on bundle deals and subscription discounts, highlighting savings on everyday essentials. These offers cater to their consistent buying habits and encourage increased basket sizes.

Big Spenders tend to purchase higher-cost items, often in categories like Electronics and Home Appliances. For this segment, high-ticket upsell opportunities and exclusive, limited-time offers are ideal. Personalized premium offers will resonate well and foster loyalty among these customers.

Recent Customers have started shopping with us but have limited purchase history. They respond best to welcome incentives such as introductory discounts or product sampling campaigns across diverse categories, helping to build their purchase frequency and brand affinity.

Lastly, Lost Customers—those who have not engaged with recent purchases—require win-back strategies involving heavy discounts and targeted email re-engagement campaigns. Special offers designed to rekindle interest can help bring these customers back into active status.

By focusing marketing efforts on these segments and tailoring offers to their demonstrated preferences, we can significantly enhance campaign ROI and customer lifetime value.

## Link Orders with Products
#### What types of offers or products are most suitable for each segment based on their past behavior?

In [112]:
# Merge orders and order items
order_items_full = order_items_clean.merge(orders_clean[['order_id', 'customer_id']], on='order_id', how='left')

# Merge with products to get product details
order_items_full = order_items_full.merge(products_clean, on='product_id', how='left')

# Merge with RFM segments
order_items_full = order_items_full.merge(rfm[['customer_id', 'segment']], on='customer_id', how='left')


In [113]:
champion_data = order_items_full[order_items_full['segment'] == 'Champions']

# Most common product categories
champion_top_categories = champion_data['category'].value_counts().head(5)
print(champion_top_categories)


category
Office Supplies    864
Electronics        776
Fashion            758
Books              744
Health             724
Name: count, dtype: int64


Based on our analysis of past customer behavior, we identified the most suitable product categories and offers for each segment. For our high-value Champions segment, we recommend focusing on Office Supplies, Electronics, and Fashion, as these categories show strong purchase activity. Tailored offers such as exclusive early access to new products, bundled deals, and loyalty program perks in these areas are likely to drive engagement and increase revenue. For the Others segment, which includes occasional and newer customers, we see significant interest in Books, Health, and Fashion categories. Introductory promotions, discounts, or free shipping on these popular items can help boost conversion and build long-term engagement. Overall, personalizing our campaign by aligning offers with these preferences should maximize impact and customer satisfaction.

In [116]:
def classify_churn(recency):
    if recency <= 30:
        return 'Active'
    elif recency <= 90:
        return 'At-Risk'
    else:
        return 'Churned'

rfm['churn_status'] = rfm['recency'].apply(classify_churn)


In [117]:
rfm['churn_status'].value_counts()


churn_status
At-Risk    993
Churned      7
Name: count, dtype: int64

In [118]:
pd.crosstab(rfm['segment'], rfm['churn_status'])


churn_status  At-Risk  Churned
segment                       
Champions          81        0
Others            912        7

In [119]:
churned_vips = rfm[(rfm['segment'].isin(['Champions', 'Big Spenders'])) & (rfm['churn_status'] == 'Churned')]
print("Churned VIPs:", len(churned_vips))


Churned VIPs: 0


We have identified early signs of customer inactivity that require our attention to minimize potential churn. Notably, 81 of our most valuable customers—classified as "Champions" for their recent, frequent, and high-value purchases—are currently in the "At-Risk" category, meaning they have not made a purchase in the past 31 to 90 days. While none of these Champions have fully churned yet (which we define as no purchase in over 90 days), this sizable group of at-risk VIP customers signals an urgent opportunity for targeted engagement.

In the broader customer base, 912 others fall into the at-risk category and 7 have fully churned. Although these customers are less critical in terms of immediate revenue impact compared to Champions, they still represent potential revenue loss if left unaddressed.

We recommend launching a personalized re-engagement campaign specifically aimed at the at-risk Champions. This could include exclusive offers such as early access to new products, limited-time discounts, or loyalty program incentives to encourage them back. Monitoring this segment closely with weekly updates will help us quickly identify any movement toward full churn.

For the smaller group of churned customers outside the Champions segment, a lighter touch campaign involving win-back emails or feedback surveys may be effective, depending on available marketing resources.

Focusing on these at-risk Champions now will maximize retention of our highest-value customers and improve overall customer lifetime value.

#### Any additional insights that could help us shape a more effective campaign?

In [121]:
# Top product categories by segment
order_items_full.groupby('segment')['category'].value_counts(normalize=True).unstack().fillna(0)



category      Books  Electronics   Fashion    Health  Home & Kitchen  \
segment                                                                
Champions  0.143022     0.149173  0.145713  0.139177        0.121300   
Others     0.145044     0.153910  0.149168  0.138833        0.119066   

category   Office Supplies    Sports  
segment                               
Champions         0.166090  0.135525  
Others            0.162131  0.131849

In [122]:
orders_clean['order_date'] = pd.to_datetime(orders_clean['order_date'], dayfirst=True)
orders_clean['day_of_week'] = orders_clean['order_date'].dt.day_name()

# Merge with RFM to get segment
orders_seg = orders_clean.merge(rfm[['customer_id', 'segment']], on='customer_id')

orders_seg.groupby('segment')['day_of_week'].value_counts(normalize=True).unstack().fillna(0)


day_of_week    Friday    Monday  Saturday    Sunday  Thursday   Tuesday  \
segment                                                                   
Champions    0.138906  0.142049  0.148963  0.127593  0.187932  0.111879   
Others       0.155234  0.150723  0.145217  0.123805  0.174123  0.097882   

day_of_week  Wednesday  
segment                 
Champions     0.142678  
Others        0.153017

In [123]:
# Delivery status by churn_status
delivery_orders = deliveries_clean.merge(orders_clean[['order_id', 'customer_id']], on='order_id')
delivery_orders = delivery_orders.merge(rfm[['customer_id', 'churn_status']], on='customer_id')

pd.crosstab(delivery_orders['delivery_status'], delivery_orders['churn_status'])


churn_status     At-Risk  Churned
delivery_status                  
Late                3574       15
On Time             3654       11

In [126]:
rfm = rfm.merge(customers_clean[['customer_id', 'has_email']], on='customer_id', how='left')

In [127]:
rfm['has_email'].value_counts()


has_email
1    950
0     50
Name: count, dtype: int64

In [128]:
vip_emailable = rfm[(rfm['segment'] == 'Champions') & (rfm['has_email'] == 1)]
print(f"We can reach {len(vip_emailable)} out of {rfm[rfm['segment'] == 'Champions'].shape[0]} Champions by email.")


We can reach 77 out of 81 Champions by email.


Based on our analysis, both Champions and Other customer segments show strong purchasing preferences for Office Supplies, Electronics, Books, and Fashion categories. These categories should be prioritized in upcoming campaigns to maximize engagement across segments. Regarding timing, Champions tend to place more orders on Thursdays and maintain steady activity on Wednesdays and Mondays, suggesting that mid-week campaign launches could be most effective for this high-value group. Other customers are more active on Mondays and Fridays, so segmenting campaign timing based on these behaviors could improve overall response rates.

Delivery performance also plays a role in customer retention. Our data indicates that customers who experience late deliveries have a slightly higher likelihood of being at risk or churning. This highlights the importance of improving shipping reliability and potentially offering compensation or apologies to those affected, which may reduce churn.

In terms of communication reach, out of 81 Champions, 77 have valid email addresses, providing a solid base for targeted email marketing. Efforts should also be made to update contact information for unreachable customers to expand this audience.

In summary, tailoring product offers to favored categories, timing campaigns to align with segment-specific ordering patterns, addressing delivery issues proactively, and focusing on email outreach for high-value customers will help shape a more effective and impactful campaign.